src/main/java/org/jabref/gui/importer/ImportEntriesDialog.java

```diff
@@ -0,0 +1,158 @@
+        this.task = task;
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(button -> {
+            if (button == importButton) {
+                viewModel.importEntries(entriesListView.getCheckModel().getCheckedItems());
+            } else {
+                dialogService.notify(Localization.lang("Import canceled"));
+            }
+
+            return null;
+        });
+    }
+
+    @FXML
+    private void initialize() {
+        viewModel = new ImportEntriesViewModel(task, taskExecutor, database, dialogService, undoManager);
+
+        Label placeholder = new Label();
+        placeholder.textProperty().bind(viewModel.messageProperty());
+        entriesListView.setPlaceholder(placeholder);
+        entriesListView.setItems(viewModel.getEntries());
+
+        PseudoClass entrySelected = PseudoClass.getPseudoClass("entry-selected");
+        new ViewModelListCellFactory<BibEntry>()
+                .withGraphic(entry -> {
+                    ToggleButton addToggle = IconTheme.JabRefIcons.ADD.asToggleButton();
+                    EasyBind.subscribe(addToggle.selectedProperty(), selected -> {
+                        if (selected) {
+                            addToggle.setGraphic(IconTheme.JabRefIcons.ADD_FILLED.withColor(IconTheme.SELECTED_COLOR).getGraphicNode());
+                        } else {
+                            addToggle.setGraphic(IconTheme.JabRefIcons.ADD.getGraphicNode());
+                        }
+                    });
+                    addToggle.getStyleClass().add("addEntryButton");
+                    addToggle.selectedProperty().bindBidirectional(entriesListView.getItemBooleanProperty(entry));
+                    HBox separator = new HBox();
+                    HBox.setHgrow(separator, Priority.SOMETIMES);
+                    Node entryNode = getEntryNode(entry);
+                    HBox.setHgrow(entryNode, Priority.ALWAYS);
+                    HBox container = new HBox(entryNode, separator, addToggle);
+                    container.getStyleClass().add("entry-container");
+                    BindingsHelper.includePseudoClassWhen(container, entrySelected, addToggle.selectedProperty());
+
+                    if (viewModel.hasDuplicate(entry)) {
+                        Button duplicateButton = IconTheme.JabRefIcons.DUPLICATE.asButton();
+                        duplicateButton.setTooltip(new Tooltip(Localization.lang("Possible duplicate of existing entry. Click to resolve.")));
+                        duplicateButton.setOnAction(event -> viewModel.resolveDuplicate(entry));
+                        container.getChildren().add(1, duplicateButton);
+                    }
+
+                    return container;
+                })
+                .install(entriesListView);
+    }
+
+    private Node getEntryNode(BibEntry entry) {
+        Node entryType = getIcon(entry.getType()).getGraphicNode();
```

Comment:
Why don't you reuse the previewPanel? It has been designed to be used for such cases.

Reply:
The PreviewPanel just displays HTML, which makes it hard to enrich the display with further functionality (such as the "more" button to expand the abstract). In the longterm I would actually propose to remove the PreviewPanel in favor of JavaFX controls.